pkg/controller/tas/topology_ungater.go

```diff
@@ -243,13 +242,21 @@ func (r *topologyUngater) Reconcile(ctx context.Context, req reconcile.Request)
 		podWithUngateInfo := &allToUngate[i]
 		var ungated bool
 		e := utilclient.Patch(ctx, r.client, podWithUngateInfo.pod, func() (bool, error) {
-			log.V(3).Info("ungating pod", "pod", klog.KObj(podWithUngateInfo.pod), "nodeLabels", podWithUngateInfo.nodeLabels)
 			ungated = utilpod.Ungate(podWithUngateInfo.pod, kueue.TopologySchedulingGate)
+			if ungated {
+				log.V(3).Info("ungating pod", "pod", klog.KObj(podWithUngateInfo.pod), "nodeLabels", podWithUngateInfo.nodeLabels)
```

Comment:
If we want to log this after ensuring we indeed send the Patch then we should log it right before returning, depending on `changed` value